lib/index-static.jsx

```diff
@@ -2,6 +2,27 @@ var React = require('react');
 
 var routes = require('./routes.jsx');
 
+// This isn't actually called in node, it's stringified and plopped in
+// a script tag in the page header. It's basically an extremely simple
+// stand-in for Modernizr, but if it becomes more complex we should think
+// about actually migrating to that library.
+//
+// Modernizr code borrowed:
+//
+// * cors (needed to reach teach-api)
+function featureDetect() {
+  var safeMode = /[?&]safemode=on/i.test(window.location.search);
+  var cors = 'XMLHttpRequest' in window &&
+             'withCredentials' in new XMLHttpRequest();
```

Comment:
Note that it's fine if this code throws an exception--that essentially means that we *definitely* want to disable JS, and because `ENABLE_JS` won't be set to `true` at the end of the function, we'll be in safe mode.